lib/ansible/modules/cloud/docker/docker_swarm_service.py

```diff
@@ -1266,6 +1306,33 @@ def has_list_changed(new_list, old_list):
     return False
 
 
+def have_networks_changed(new_networks, old_networks):
+    """Special case list checking for networks to sort aliases"""
+
+    if new_networks is None:
+        return False
+    old_networks = old_networks or []
+    if len(new_networks) != len(old_networks):
+        return True
+
+    zip_data = zip(
+        sorted(new_networks, key=lambda k: k['id']),
+        sorted(old_networks, key=lambda k: k['id'])
+    )
+
+    for new_item, old_item in zip_data:
+        # Sort the aliases
```

Comment:
Please create shallow copies to avoid the original data to be modified:
```suggestion
        new_item = dict(new_item)
        old_item = dict(old_item)
        # Sort the aliases
```